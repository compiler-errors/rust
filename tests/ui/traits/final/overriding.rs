#![feature(final_associated_functions)]

trait Foo {
    final fn method() {}
}

impl Foo for () {
    fn method() {}
    //~^ ERROR cannot override `method` because it already has a `final` definition in the trait
}

fn main() {}
