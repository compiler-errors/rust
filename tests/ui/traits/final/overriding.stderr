error: cannot override `method` because it already has a `final` definition in the trait
  --> $DIR/overriding.rs:8:5
   |
LL |     fn method() {}
   |     ^^^^^^^^^^^
   |
note: `method` is marked final here
  --> $DIR/overriding.rs:4:5
   |
LL |     final fn method() {}
   |     ^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

